src/download-pool.js

```diff
@@ -2,38 +2,75 @@ import downloadFile from './download-file';
 
 const POOL_LIMIT = 6;
 
+/**
+ * Groups the elements in an array in subarrays defined by limit.
+ * @example
+ *
+ * splice([1, 2, 3, 4, 5], 3); // [[1, 2, 3], [4, 5]]
+ *
+ * @param {Array} arr
+ * @param {Number} limit
+ * @returns {Array} New array with the subgroups.
+ */
 export const splice = (arr, limit) => {
   const splicedArray = [];
+  const baseArray = arr.slice(0);
```

Comment:
The problem with that kind of helpers is that they can be hard to remember on time, but for this case we can do it if you want. I don't mind. For the `fs.stat` it makes a lot of sense because you're right. It is really not telling anything about our purpose with it.